src/dal_select2/views.py

```diff
@@ -44,3 +45,47 @@ def render_to_response(self, context):
 
 class Select2QuerySetView(Select2ViewMixin, BaseQuerySetView):
     """List options for a Select2 widget."""
+
+class Select2ListView(ViewMixin, View):
+    """Autocomplete from a list of items rather than a QuerySet.
+    get_list() should return a list of strings that may be autocopleted.
+    """
+
+    create_field = False
+    """Set create_field to True to allow creation of new items."""
+
+    def get_list(self):
+        """"Return the list strings from which to autocomplete."""
+        return []
+
+    def get(self, request, *args, **kwargs):
+        results = self.get_list()
+        create_option = []
+        if self.q:
+            results = [ x for x in results if self.q in x ]
+            if self.create_field:
+                create_option = [{
+                        'id': self.q,
+                        'text': 'Create "%s"' % self.q,
+                        'create_id': True
+                    }]
+        return http.HttpResponse(json.dumps({
+            'results': [dict(id=x, text=x) for x in results] + create_option
+        }))
+
+    def create(self, text):
+        """"Create a new list item for text.  None can be returned if the
+        item may not be created for some reason."""
+        return text
```

Comment:
I think the verb "create" implies some new object is persisted somewhere. Perhaps you could describe why that works for as-is ?